EIPS/eip-778.md

```diff
@@ -0,0 +1,104 @@
+# Preamble
+
+    EIP: 778
+    Title: Ethereum Node Records (ENR)
+    Author: Felix Lange <fjl@ethereum.org>
+    Type: Standard Track
+    Category Networking
+    Status: Draft
+    Created: 2017-11-23
+
+# Abstract
+
+This EIP defines Ethereum Node Records, an open format for p2p connectivity information.
+
+# Motivation
+
+Ethereum nodes discover each other through the node discovery protocol. The purpose of
+that protocol is relaying node identity public keys (on the secp256k1 curve), their IP
+address and two port numbers. No other information can be relayed.
+
+This specification seeks to lift the restrictions of the discovery v4 protocol by defining
+a flexible format, the *node record*, for connectivity-related information. Node records
+can be relayed through a future version of the node discovery protocol. They can also be
+relayed through arbitrary other mechanisms such as DNS, ENS, a devp2p subprotocol, etc.
+
+Node records improve cryptographic agility and handling of protocol upgrades. A record can
+contain information about arbitrary transport protocols and public key material associated
+with them.
+
+Another goal of the new format is to provide authoritative updates of connectivity
+information. If a node changes its endpoint and publishes a new record, other nodes should
+be able to determine which record is newer.
+
+# Specification
+
+The components of a node record are:
+
+- `signature`: cryptographic signature of record contents
+- `seq`: The sequence number, a 64 bit integer. Nodes should increase the number whenever
+   the record changes and republish the record.
+-  The remainder of the record consists of arbitrary key/value pairs, which must be sorted
+   by key.
+
+A record's signature is made and validated according to an *identy scheme*. The identity
```

Comment:
small typo - should be "identity scheme"

Reply:
fixed